app/src/main/AndroidManifest.xml

```diff
@@ -80,7 +80,7 @@
         <activity
             android:name=".activity.MainActivity"
             android:configChanges="keyboardHidden|orientation|screenSize|smallestScreenSize|screenLayout|density|uiMode|keyboard|navigation"
-            android:windowSoftInputMode="stateHidden"
+            android:windowSoftInputMode="stateAlwaysHidden"
```

Comment:
The  [documentation](https://developer.android.com/guide/topics/manifest/activity-element#wsoft) says:
` "stateAlwaysHidden" 	The soft keyboard is always hidden when the activity's main window has input focus.`

As all search actions, or actions where the user has to input some text, were handled by fragments, the `activity's main window` does not get the focus and so the keyboard does not hide.
Anyway after dismissing a dialog the main window gets the focus again and the keyboard will hide.